install/infra/single-cluster/gcp/README.md

```diff
@@ -0,0 +1,235 @@
+
+## Note the NS records from terraform output
+
+Once the apply process has exited successfully, we can go ahead and prepare to
+setup Gitpod. If you specified the `domain_name` in the `terraform.tfvars` file,
+the terraform module registers the module with `cloudDNS` to point to the
+cluster. Now you have to configure whichever provider you use to host your
+domain name to route traffic to the GCP name servers. You can find these name
+servers in the `make output` command from above. It would be of the format:
+
+```json
+Nameservers for the domain(to be added as NS records in your domain provider):
+=================
+[
+  "ns-cloud-c1.googledomains.com.",
+  "ns-cloud-c2.googledomains.com.",
+  "ns-cloud-c3.googledomains.com.",
+  "ns-cloud-c4.googledomains.com."
+]
+
+
+```
+
+Add the `ns` records similar to the above 4 URIs as NS records under your domain
+DNS management setup. Check with your domain hosting service for specific information.
+
+## Note the dependency credentials from terraform output
+
+If you enabled the creation of external database, storage and registry, the
+above `make output` command would list the credentials to connect to the
+corresponding resource. Make a note of this, so as to provide the same when
+setting up Gitpod via KOTS UI.
+
+## Install Gitpod
+
+You can install `KOTS` CLI to install Gitpod:
+
+``` sh
+curl https://kots.io/install | bash
+```
+
+Run the following to get started with Gitpod installation:
+
+``` sh
+export KUBECONFIG=kubeconfig
+kubectl kots install gitpod/stable # you can replace `stable` with `unstable` or `beta` as per the requirement
+```
+
+Upon completion, you can access `KOTS` UI in `localhost:8800`. Here you can
+proceed to configuring and intalling Gitpod. Please follow the [official
+documentaion](https://www.gitpod.io/docs/self-hosted/latest/getting-started#step-4-install-gitpod) to complete the Gitpod setup.
+
+## Troubleshooting
+
+### Some pods never start (Init state)
+
+```sh
+kubectl get pods -l component=proxy
+NAME                     READY   STATUS    RESTARTS   AGE
+proxy-5998488f4c-t8vkh   0/1     Init 0/1  0          5m
+```
+
+The most likely reason is that the DNS01 challenge has yet to resolve. To fix this, make sure you have added the NS records corresponding to the `cloudDNS` zone of the `domain_name` added to your domain provider.
+
+Once the DNS record has been updated, you will need to delete all Cert Manager pods to retrigger the certificate request
+
+```
+kubectl delete pods -n cert-manager --all
+```
+
+After a few minutes, you should see the https-certificate become ready.
+
+```
+kubectl get certificate
+NAME                        READY   SECRET                      AGE
+https-certificates          True    https-certificates          5m
+```
+
+### Cannot connect to the created cluster after a while
+
+There is a chance that your kubeconfig has gotten expired after a specific amount of time. You can reconnect to the cluster by using:
```

Comment:
I typically encounter this when I'm reopening a Gitpod workspace; upon opening that workspace my `~/.kube/config` credentials are lost between workspace instances. Is this the scenario that we're looking to support? Are there other potential sources for lost kube credentials?

Reply:
So as long as one person has access to the service account creadentials and the name and region of the cluster, they can run the `gcloud container clusters` command mentioned here. I will append the command to activate the service account too before this. Would that address the issue you mentioned?